src/vs/workbench/contrib/tasks/browser/runAutomaticTasks.ts

```diff
@@ -44,13 +50,16 @@ export class RunAutomaticTasks extends Disposable implements IWorkbenchContribut
 			this._logService.trace('RunAutomaticTasks: Awaiting task system info.');
 			await Event.toPromise(Event.once(this._taskService.onDidChangeTaskSystemInfo));
 		}
-
+		const isWorkspaceTrusted = this._workspaceTrustManagementService.isWorkspaceTrusted();
+		if (this._hasRunTasks || !isWorkspaceTrusted || this._configurationService.getValue(ALLOW_AUTOMATIC_TASKS) === 'off') {
+			return;
+		}
 		const workspaceTasks = await this._taskService.getWorkspaceTasks(TaskRunSource.FolderOpen);
 		this._logService.trace(`RunAutomaticTasks: Found ${workspaceTasks.size} automatic tasks`);
-		await RunAutomaticTasks.runWithPermission(this._taskService, this._storageService, this._notificationService, this._workspaceTrustManagementService, this._openerService, this._configurationService, workspaceTasks);
+		await this._runWithPermission(this._taskService, this._storageService, this._notificationService, this._workspaceTrustManagementService, this._openerService, this._configurationService, workspaceTasks);
```

Comment:
I think we want to set `this._hasRunTasks = true;` immediately after this if, otherwise `_tryRunTasks` could potentially be run again before `this._taskService.getWorkspaceTasks` has resolved.